pkg/gardenlet/controller/backupentry/migration/add.go

```diff
@@ -0,0 +1,116 @@
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+)
+
+// ControllerName is the name of this controller.
+const ControllerName = "backupentry-migration"
+
+// AddToManager adds Reconciler to the given manager.
+func (r *Reconciler) AddToManager(mgr manager.Manager, gardenCluster cluster.Cluster) error {
+	if r.GardenClient == nil {
+		r.GardenClient = gardenCluster.GetClient()
+	}
+
+	// It's not possible to overwrite the event handler when using the controller builder. Hence, we have to build up
+	// the controller manually.
+	c, err := controller.New(
+		ControllerName,
+		mgr,
+		controller.Options{
+			Reconciler:              r,
+			MaxConcurrentReconciles: pointer.IntDeref(r.Config.Controllers.BackupEntryMigration.ConcurrentSyncs, 0),
+			RecoverPanic:            true,
+			// if going into exponential backoff, wait at most the configured sync period
+			RateLimiter: workqueue.NewWithMaxWaitRateLimiter(workqueue.DefaultControllerRateLimiter(), r.Config.Controllers.BackupEntryMigration.SyncPeriod.Duration),
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	return c.Watch(
+		source.NewKindWithCache(&gardencorev1beta1.BackupEntry{}, gardenCluster.GetCache()),
+		&handler.EnqueueRequestForObject{},
+		&predicate.GenerationChangedPredicate{},
+		r.IsBeingMigratedPredicate(gardenCluster.GetClient()),
+	)
+}
+
+// IsBeingMigratedPredicate returns a predicate which returns true for backup entries that are being migrated to a different seed.
+func (r *Reconciler) IsBeingMigratedPredicate(reader client.Reader) predicate.Predicate {
+	return &isBeingMigratedPredicate{
+		reader:   reader,
+		seedName: r.Config.SeedConfig.Name,
+	}
+}
+
+type isBeingMigratedPredicate struct {
+	ctx      context.Context
+	reader   client.Reader
+	seedName string
+}
+
+func (p *isBeingMigratedPredicate) InjectStopChannel(stopChan <-chan struct{}) error {
+	p.ctx = contextutil.FromStopChannel(stopChan)
+	return nil
+}
+
+func (p *isBeingMigratedPredicate) Create(e event.CreateEvent) bool {
+	return p.isBeingMigratedToSeed(e.Object)
+}
+func (p *isBeingMigratedPredicate) Update(e event.UpdateEvent) bool {
+	return p.isBeingMigratedToSeed(e.ObjectNew)
+}
+func (p *isBeingMigratedPredicate) Delete(e event.DeleteEvent) bool {
+	return p.isBeingMigratedToSeed(e.Object)
+}
+func (p *isBeingMigratedPredicate) Generic(e event.GenericEvent) bool {
+	return p.isBeingMigratedToSeed(e.Object)
+}
+
+func (p *isBeingMigratedPredicate) isBeingMigratedToSeed(obj client.Object) bool {
+	backupEntry, ok := obj.(*gardencorev1beta1.BackupEntry)
+	if !ok {
+		return false
+	}
+	if backupEntry.Spec.SeedName != nil && backupEntry.Status.SeedName != nil && *backupEntry.Spec.SeedName != *backupEntry.Status.SeedName && *backupEntry.Spec.SeedName == p.seedName {
+		seed := &gardencorev1beta1.Seed{}
+		if err := p.reader.Get(p.ctx, kutil.Key(*backupEntry.Status.SeedName), seed); err != nil {
+			return false
+		}
+		return v1beta1helper.SeedSettingOwnerChecksEnabled(seed.Spec.Settings)
```

Comment:
Where is this coming from? 😕 Can you point me the respective code in the `master` branch?

Reply:
https://github.com/gardener/gardener/blob/b9a7250a02230b801312f6cdf89730e1bf1677b4/pkg/controllerutils/seedfilter.go#L105-L128